### PR TITLE
Broken Link Checker

### PR DESCRIPTION
**Pull request by kshitjee**
### Description
Add github workflow to fail in CI if broken links are detected in any HTML or Markdown files under apps/nextra. 

### Checklist

- Do all Lints pass?
  - [] Have you ran `pnpm spellcheck`?
  - [] Have you ran `pnpm fmt`?
  - [] Have you ran `pnpm lint`?

### CODE REVIEW

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches:
+      - main
+      - kc/link-checker
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron: "00 18 * * *"  # This will run the check daily at 18:00 UTC
```
Comment (hariria) [inline .github/workflows/link-check.yml:12]: Let's schedule this to run when someone opens a PR, not dailly.

File: .github/workflows/link-check.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Run Lychee Link Checker
+      uses: lycheeverse/lychee-action@v1
+      with:
+        args: --base . --verbose --no-progress './apps/nextra/**/*.mdx' './apps/nextra/**/*.html'
```
Comment (gregnazario) [inline .github/workflows/link-check.yml:19]: We probably need to figure out how to ignore `localhost` links.

This is why the broken link checker wasn't blocking before

Comment (hariria) [reply .github/workflows/link-check.yml:19]: Can you give an example of what you mean by a local host link (I know what it is but when would it arise in the docs?)

Comment (gregnazario) [reply .github/workflows/link-check.yml:19]: Just take a look at the output from the PR run

Comment (hariria) [reply .github/workflows/link-check.yml:19]: Oh lmfao i see thanks. @kshitjee maybe something like this https://github.com/lycheeverse/lychee/issues/1384#issuecomment-1970108569

File: .github/workflows/link-check.yml (hunk 1 above)
Comment (kshitjee) [reply .github/workflows/link-check.yml]: done

File: .github/workflows/link-check.yml (hunk 2 above)
Comment (kshitjee) [reply .github/workflows/link-check.yml]: I think it ignores localhost links now, check latest workflow run

**Review by gregnazario — APPROVED** [review]

**Review by hariria — APPROVED** [review]

**Pull request merged**